daemon/cluster/executor/container/controller.go

```diff
@@ -321,6 +323,13 @@ func (r *controller) Shutdown(ctx context.Context) error {
 		return err
 	}
 
+	// add a delay for gossip converge
+	// TODO(dongluochen): this delay shoud be configurable to fit different cluster size and network delay.
+	// TODO(dongluochen): if there is no service binding for this container, this delay is not necessary.
+	// if r.adapter.deactivateServiceBinding can return an error to indicate that, it'd reduce service
+	// converge time.
```

Comment:
Whether the delay should be added or not is something that has to be decided by the caller of Shutdown (may be based on user input ?) and not based on the return value from deactivateServiceBinding(). We will soon be supporting local drivers in swarm mode. Those tasks can also expose a service but don't need the delay because there is no propagation outside the host involved in that case.